plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -1040,6 +1069,9 @@ def _identify_wanted_versions(self, package_info_generator):
             for unit, info in units.itervalues():
                 ret[unit] = info
 
+        if package_type == ids.TYPE_ID_RPM:
```

Comment:
The idea was to check and raise those exceptions at a reasonably high level and closer to each other, so the metadata check will happen at almost the same level. This way it may be easier (for the person who will convert this functionality to proper metadata validation in Pulp 3 later) to get the idea what to check, rather than looking for various small pieces buried deep down in seemingly unrelated functions/methods.
That's the main reason but maybe not a good idea.